oxidation/libparsec/crates/types/src/regex.rs

```diff
@@ -1,8 +1,27 @@
 // Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 (eventually AGPL-3.0) 2016-present Scille SAS
 
-use std::fmt::Display;
+use std::{
+    collections::HashSet,
+    fmt::{Display, Write},
+    fs,
+    path::Path,
+};
 
-pub struct Regex(pub regex::Regex);
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum RegexError {
+    #[error("Regex parsing err: {err}")]
+    ParseError { err: regex::Error },
+    #[error("Failed to convert glob pattern into regex: {err}")]
+    GlobPatternError { err: fnmatch_regex::error::Error },
+    #[error("Failed to open pattern file: {err}")]
+    PatternFileIOError { err: std::io::Error },
+}
+
+pub type RegexResult = Result<Regex, RegexError>;
+
+pub struct Regex(pub Vec<regex::Regex>);
```

Comment:
The name isn't correct

Reply:
Do you have any suggestion? I think `Regex` is still a good fit here as all it does it matching patterns. The only difference is that it can be built from glob expressions.

Reply:
We could rename `RegexSet` but it may be confusing with `regex::RegexSet` as it isn't the same behavior.